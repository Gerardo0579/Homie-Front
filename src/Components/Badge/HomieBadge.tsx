import React, { FC } from 'react'
import styles from './HomieBadge.module.css'

interface HomieBadgeProps {
  content: string | number
}

export const HomieBadge: FC<HomieBadgeProps> = ({ content }) => {
  return <div className={styles.badgeTypeRepo}>{content}</div>
}
