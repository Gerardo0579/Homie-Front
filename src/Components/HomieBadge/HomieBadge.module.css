.badgeTypeRepo{
    display: inline-block;
    border: 1px solid #d0d7de;
    border-radius: 2em;
    padding: 0 7px;
    font-size: 12px;
    font-weight: 500;
    width: auto;
    color: #57606a;
}