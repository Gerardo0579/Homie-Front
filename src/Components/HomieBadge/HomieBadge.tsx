import React, { FC } from 'react'
import styles from './HomieBadge.module.css'

interface HomieBadgeProps {
  content: string | number
  backgroundColor: string
  borderColor: string
  fontColor: string
}

export const HomieBadge: FC<HomieBadgeProps> = ({
  content,
  backgroundColor,
  borderColor,
  fontColor
}) => {
  return (
    <div
      style={{
        borderColor: borderColor,
        backgroundColor: backgroundColor,
        color: fontColor
      }}
      className={styles.badgeTypeRepo}>
      {content}
    </div>
  )
}
