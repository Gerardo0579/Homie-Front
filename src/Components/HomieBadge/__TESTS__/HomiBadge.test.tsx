import renderer from 'react-test-renderer'
import { HomieBadge } from '../HomieBadge'

const CONTENT: string = '1'
const BACKGROUND_COLOR: string = '#fff'
const BORDER_COLOR: string = '#AAA'
const FONT_COLOR: string = '#000'

it('renders correctly', () => {
  const tree = renderer
    .create(
      <HomieBadge
        content={CONTENT}
        backgroundColor={BACKGROUND_COLOR}
        borderColor={BORDER_COLOR}
        fontColor={FONT_COLOR}
      />
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
