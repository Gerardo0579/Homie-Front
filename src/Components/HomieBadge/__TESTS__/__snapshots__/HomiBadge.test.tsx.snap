// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="badgeTypeRepo"
  style={
    Object {
      "backgroundColor": "#fff",
      "borderColor": "#AAA",
      "color": "#000",
    }
  }
>
  1
</div>
`;
